oxide-vpc/src/engine/dhcp.rs

```diff
@@ -39,6 +39,13 @@ pub fn setup(
     cfg: &VpcCfg,
     ft_limit: core::num::NonZeroU32,
 ) -> Result<(), OpteError> {
+    // The DHCP layer only contains meaningful actions if the port is configured
+    // to support IPv4.
+    let ip_cfg = match cfg.ipv4_cfg() {
+        None => return Ok(()),
```

Comment:
The idea of a "gateway" layer that acts that way is definitely appealing. It centralizes the logic for deciding which packets make it anywhere beyond the guest. At the same time, having layers for each protocol has definitely been helpful, for testing and development. It's nice to be able to see just ICMP or whatever, for example when using `opteadm dump-layer`.

I've been thinking that it would be very nice to name rules within layers, even before you bring this up. This would I think make that more valuable, since one could read a name for the rule, say `icmpv6-guest-echo-reply`, and see the actions and predicates associated with that. It gets a bit funky with rules that are related, like the inbound and outbound rules to drop ARP, but I still think it might be helpful.